services/director-v2/src/simcore_service_director_v2/modules/osparc_variables/substitutions.py

```diff
@@ -69,6 +70,73 @@ async def substitute_vendor_secrets_in_model(
     return result
 
 
+async def substitute_vendor_secrets_in_specs(
+    app: FastAPI,
+    specs: dict[str, Any],
+    *,
+    safe: bool = True,
+    service_key: ServiceKey,
+    service_version: ServiceVersion,
+    product_name: ProductName,
+) -> dict[str, Any]:
+    resolver = SpecsSubstitutionsResolver(specs, upgrade=False)
+    repo = get_repository(app, ServicesEnvironmentsRepository)
+
+    _logger.debug(
+        "substitute_vendor_secrets_in_specs detected_identifiers=%s",
+        resolver.get_identifiers(),
+    )
+
+    if any(repo.is_vendor_secret_identifier(idr) for idr in resolver.get_identifiers()):
+        # checks before to avoid unnecessary calls to pg
+        vendor_secrets = await repo.get_vendor_secrets(
+            service_key=service_key,
+            service_version=service_version,
+            product_name=product_name,
+        )
+
+        # resolve substitutions
+        resolver.set_substitutions(mappings=vendor_secrets)
+        new_specs: dict[str, Any] = resolver.run(safe=safe)
+        return new_specs
+
+    return deepcopy(specs)
+
+
+class OsparcSessionVariablesTable(OsparcVariablesTable, SingletonInAppStateMixin):
+    app_state_name: str = "session_variables_table"
+
+    @classmethod
+    def create(cls, app: FastAPI):
+        table = cls()
+        # Registers some session osparc_variables
+        # WARNING: context_name needs to match session_context!
+        # NOTE: please keep alphabetically ordered
+        for name, context_name in [
+            ("OSPARC_VARIABLE_NODE_ID", "node_id"),
+            ("OSPARC_VARIABLE_PRODUCT_NAME", "product_name"),
+            ("OSPARC_VARIABLE_STUDY_UUID", "project_id"),
+            ("OSPARC_VARIABLE_USER_ID", "user_id"),
+            ("OSPARC_VARIABLE_API_HOST", "api_server_base_url"),
+        ]:
+            table.register_from_context(name, context_name)
+
+        table.register_from_handler("OSPARC_VARIABLE_USER_EMAIL")(request_user_email)
+        table.register_from_handler("OSPARC_VARIABLE_USER_ROLE")(request_user_role)
+        table.register_from_handler("OSPARC_VARIABLE_API_KEY")(
+            get_or_create_user_api_key
+        )
+        table.register_from_handler("OSPARC_VARIABLE_API_SECRET")(
+            get_or_create_user_api_secret
+        )
```

Comment:
I have one issue with this. It lowers securtity. I did not see any mention of this in the body of the PR.
Could you please trace it in an issue.
Security goes down becase: all services will now start with the same key. A key will not be created and removed each time the service is started and shut down.

FYI: @elisabettai @mguidon @sanderegg

Reply:
@GitHK As we agreed in the offline peer review, I will follow up on this in https://github.com/ITISFoundation/osparc-simcore/issues/5875 (as referred in the description)